paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/FlinkSourceBuilder.java

```diff
@@ -142,7 +144,10 @@ private ContinuousFileStoreSource buildContinuousFileSource() {
 
             // TODO visit all options through CoreOptions
             StartupMode startupMode = CoreOptions.startupMode(conf);
-            if (logSourceProvider == null) {
+            StreamReadType streamReadType = CoreOptions.streamReadType(conf);
+
+            // no log system or
+            if (logSourceProvider == null || streamReadType == FILE) {
```

Comment:
When `logSourceProvider` isn't null and `streamReadType` is `FILE`, is it right to build continuous file source like `latest` startup mode?

Reply:
no log system or log system is configured but streamReadType is FILE

Reply:
no log system or log system is configured but streamReadType is FILE